config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1878,6 +1878,19 @@ public HttpBasicSpec securityContextRepository(ServerSecurityContextRepository s
 			return this;
 		}
 
+		/**
+		 * Allows easily changing the realm.
+		 * @param realmName the HTTP Basic realm to use
+		 * @return {@link HttpBasicSpec} for additional customization
+		 * @since 5.2.0
+		 * @author Ankur Pathak
+		 */
+		public HttpBasicSpec realmName(String realmName){
```

Comment:
I think that rather than specifying realmName we should allow injection of the entryPoint instead. This gives the user more flexibility and we don't need this additional logic which does an instance of check.

Reply:
I do agree, allowing setting entryPoint will give user more flexibility. So method to set realmName
removed in its favour. The new method authenticationEntryPoint addedin its place.